projects/plugins/wpcomsh/custom-colors/colors.php

```diff
@@ -400,7 +400,7 @@ public static function register_scripts_and_styles() {
 	 * @param array $classes the array of classes to add custom class to.
 	 */
 	public static function body_class( $classes ) {
-		array_push( $classes, 'custom-colors' );
```

Comment:
on this should we do an explicit check if $classes is an array? if an unexpected value set it to an array.

Either way thanks for putting in the time to clear up these items.

Reply:
With `$classes[] = 'string';`, if `$classes` is `null` then PHP will silently convert it into an empty array. Since `null` is the only invalid incoming value we're seeing, that should be fine to resolve that.

As for more extensive checks, WordPress Core passes an array, and immediately uses the returned value as an array, and documents that the value is supposed to be an array. Whatever filter is returning a non-array is already broken even without Jetpack involved. In theory we might have _every_ function have a bunch of boilerplate at the top to check the types of its variables and do something if it gets invalid values, although most of the time the "something" would probably be to throw an InvalidArgumentException. But is doing that worth the effort? Occasionally yes, but here I think probably not.